docs/DeveloperGuide.md

```diff
@@ -1,34 +1,890 @@
+##### Implementation
+When the user attempts to delete the `Chicken Rice` recipe from Kitchen Helper, the `Kitchen Helper`, `Parser` and `DeleteRecipeCommand` class will be called upon. The following sequence of steps will then occur: 
+1. The user keyed in “deleterecipe /n `Chicken Rice`".
+    2. A `UI` object will be created and it will call `UI#getUserCommand()` method to take in the input that the user has keyed in. 
+    3. A `String` object will be returned and saved into the `userCommandInput` variable in `Kitchen Helper`. 
+    4. The variable `userCommandInput` is being parsed into the `Parser` class as an argument for this method `Parser#parseUserCommand()`.
+2. The command inserted by the user is being parsed into the `Parser` and a new `Command` object is being created. 
+    2. The variable `userCommandInput` will be identified as `deleterecipe` in the `Parser#parseUserCommand()`.The `Parser#prepareDeleteRecipe()` is being called to prepare the `userCommandInput` string to create a `DeleteRecipeCommand` object.
+3. The command is now being executed.
+    2. The `DeleteRecipeCommand#execute()` will be called.
+    3. As this is a deletion by recipe name, the `recipeIndex` variable is set as null. As the variable is null, `DeleteRecipeCommand#deleteRecipeByName()` will be called.
+    4. Next, the `DeleteRecipeCommand#getRecipeIndex()` to get the index based on the recipe name that the user has inputted. With the given index, `DeleteRecipeCommand#deleteRecipe()` will be called to delete the recipe. 
+    5. Lastly, a String called `feedbackToUser` will be returned to the user to inform the user of the outcome of the command. 
+4. The details will then be printed onto the console using `Ui#showResultToUser(result)`.
+
+
+##### Design Considerations
+Aspect: How is the `DeleteRecipeCommand` initialise. <br>
+<br>
+Alternative 1 (Current Choice): Usage of 2 constructors <br>
+
+|     |     |
+|-----|-----|
+|**Pros** | This gives us more flexibility on what object can be created with different variables since there are two methods of recipe deletion. |  
+|**Cons** | There is an overload of constructors.|
+
+Rationale for using this: As there are two different types of deletion, it would be simpler and increase cohesion as it is more easier to express these constructors' functionality at a higher level.
+
+Alternative 2: Usage of 1 constructor <br>
+
+|     |     |
+|-----|-----|
+|**Pros** |The Parser can call for one main default constructor. |
+|**Cons** | The single constructor will need to deal with 2 different methods of deletion, causing the constructor to have more than one purpose.|
+
+
+#### 4.2.5. Search for recipe based on keyword(s)
+
+The search for recipe feature allows the user to find recipes using a keyword in the recipe’s list.  
+For example, `searchrecipe Chicken` will find all recipes that contain `Chicken`.  
+
+
+##### Implementation
+
+![SearchRecipeCommand](images/searchRecipeCommand1.png)
```

Comment:
Could potentially expand to show sequence diagram for "add to findRecipeList" and "add to recipeIndex" in the opt frame of the loop of the SearchRecipeCommand. Possibly in a separate diagram, since this is complicated enough